apps/prairielearn/src/schemas/schemas/infoCourse.ts

```diff
@@ -0,0 +1,172 @@
+import { z } from 'zod';
+
+export const ColorSchema = z
+  .enum([
+    'red1',
+    'red2',
+    'red3',
+    'pink1',
+    'pink2',
+    'pink3',
+    'purple1',
+    'purple2',
+    'purple3',
+    'blue1',
+    'blue2',
+    'blue3',
+    'turquoise1',
+    'turquoise2',
+    'turquoise3',
+    'green1',
+    'green2',
+    'green3',
+    'yellow1',
+    'yellow2',
+    'yellow3',
+    'orange1',
+    'orange2',
+    'orange3',
+    'brown1',
+    'brown2',
+    'brown3',
+    'gray1',
+    'gray2',
+    'gray3',
+  ])
+  .describe('A color name.');
+
+export type Color = z.infer<typeof ColorSchema>;
+export const TopicSchema = z
+  .object({
+    comment: z
+      .union([z.string(), z.array(z.any()), z.object({}).catchall(z.any())])
+      .describe('Arbitrary comment for reference purposes.')
+      .optional(),
+    shortName: z.string().describe('Short name (preferably 2 to 7 characters).').optional(),
+    name: z.string().describe('Long descriptive name (preferably less than 10 words).'),
+    color: ColorSchema,
+    description: z.string().describe('Description of the topic.').optional(),
+  })
+  .describe('A single assessment set description.');
+
+export type Topic = z.infer<typeof TopicSchema>;
+export const AssessmentSetSchema = z
+  .object({
+    comment: z
+      .union([z.string(), z.array(z.any()), z.object({}).catchall(z.any())])
+      .describe('Arbitrary comment for reference purposes.')
+      .optional(),
+    abbreviation: z
+      .string()
+      .describe("Abbreviation (preferably 1 to 3 characters), e.g., 'HW', 'Q', 'PQ', etc."),
+    name: z
+      .string()
+      .describe(
+        "Full singular name (preferably 1 to 3 words), e.g., 'Homework', 'Quiz', 'Practice Quiz'.",
+      ),
+    heading: z
+      .string()
+      .describe(
+        "Plural heading for a group of assessments (preferably 1 to 3 words), e.g., 'Homeworks', 'Quizzes'.",
+      ),
+    color: ColorSchema,
+  })
+  .describe('A single assessment set description.');
+
+export type AssessmentSet = z.infer<typeof AssessmentSetSchema>;
```

Comment:
Same comment re: this type having the same name as in `db-types.ts`. Consider names like `CourseJsonAssessmentSetSchema`? Ditto for other files, I won't comment on all things we might want to rename yet.

Reply:
I like the naming schema `AssessmentSetJson`  / `AssessmentSetJsonSchema`.

Reply:
We should probably agree on a naming scheme before I rename, as it will be a little tedious.

Reply:
I want `Schema` at the end, and the same prefix for both. I would prefer doing `Json` after the name, instead of as a prefix.

Reply:
I'm good with `*JsonSchema` for schemas and `*Json` for types.

Reply:
Changed in [e36a8e5](https://github.com/PrairieLearn/PrairieLearn/pull/11024/commits/e36a8e5af108dfed07f45694f74dd9ec0d5407fc).